influxdb3_write/src/last_cache.rs

```diff
@@ -802,22 +888,47 @@ impl LastCacheStore {
     /// for the cache.
     fn to_record_batch(
         &self,
+        output_schema: &ArrowSchemaRef,
         extended: Option<(Vec<FieldRef>, Vec<ArrayRef>)>,
     ) -> Result<RecordBatch, ArrowError> {
         // This is where using IndexMap is important, because we inserted the cache entries
         // using the schema field ordering, we will get the correct order by iterating directly
         // over the map here:
-        let cache_arrays = self.cache.iter().map(|(_, c)| c.data.as_array());
-        let (schema, arrays) = if let Some((fields, mut arrays)) = extended {
-            let mut sb = ArrowSchemaBuilder::new();
-            sb.extend(fields);
-            sb.extend(self.schema.fields().iter().map(Arc::clone));
-            arrays.extend(cache_arrays);
-            (Arc::new(sb.finish()), arrays)
-        } else {
-            (Arc::clone(&self.schema), cache_arrays.collect())
-        };
-        RecordBatch::try_new(schema, arrays)
+        let (fields, mut arrays): (Vec<FieldRef>, Vec<ArrayRef>) =
```

Comment:
Some code comments would be helpful here to explain the condition and use of skip below.